api/hpa.go

```diff
@@ -253,8 +282,10 @@ func DeleteDeploymentAutoscalingInfo(kubeConfig []byte, scaleTarget string) erro
 	// get doesn't work with v12.NamespaceAll only if you specify the namespace exactly
 	// deployment, err := client.AppsV1().Deployments(v12.NamespaceAll).Get(request.Name, v12.GetOptions{})
 	scaleTargetFound := false
-
-	deploymentList, err := client.AppsV1().Deployments(v12.NamespaceAll).List(v12.ListOptions{})
+	listOptions := v12.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name=%v", scaleTarget),
+	}
+	deploymentList, err := client.AppsV1().Deployments(v12.NamespaceAll).List(listOptions)
```

Comment:
Please handle the error here

Reply:
I just wondering how does this went through our lint mechanism, it suppose to fail with something "error does not handled properly" cc @sagikazarmark

Reply:
I suppose it did, because the err variable is actually used later. That's why `ineffassign` lint didn't kick in. The `errcheck` lint is explicitly disabled. 😄

Reply:
fixed